pkg/cloud/aws/services/userdata/userdata.go

```diff
@@ -90,3 +93,21 @@ func funcMap(funcs map[string]interface{}) template.FuncMap {
 
 	return funcMap
 }
+
+func GetUserDataFromSecret(machine *actuators.MachineScope) (string, error) {
```

Comment:
A couple of suggestions and questions:
* flip around the nil check and return early if the UserDataSecret is nil
* the `userData` var is unnecessary, directly encode and return at line 105 and remove the else statement
* why is it an error if the secret doesn't contain the `userData` key? My expectation would be to log an error and return nothing
* please add godoc and a test to this function